### PR TITLE
Core fix retry

### PR DESCRIPTION
**Pull request by xiangyan99**

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/core/azure-core/azure/core/pipeline/policies/_retry.py [hunk 1]
```diff
@@ -46,6 +47,9 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+class RetryType(Enum):
```
Comment (johanste) [inline sdk/core/azure-core/azure/core/pipeline/policies/_retry.py]: I would use:

```python
class RetryType(str, Enum):
    Exponential = 'exponential'
    Fixed = 'fixed'
```

File: sdk/core/azure-core/azure/core/pipeline/policies/_retry.py [hunk 2]
```diff
@@ -73,6 +77,8 @@ class RetryPolicy(HTTPPolicy):
 
     :keyword int retry_backoff_max: The maximum back off time. Default value is 120 seconds (2 minutes).
 
+    :keyword RetryType retry_type: Fixed or exponential delay between retries, default is exponential.
```
Comment (johanste) [inline sdk/core/azure-core/azure/core/pipeline/policies/_retry.py]: The docstring for `retry_backoff` must be updated to reflect its behavior for fixed backoff...

Comment (johanste) [inline sdk/core/azure-core/azure/core/pipeline/policies/_retry.py]: Name of parameter: `retry_mode`

File: sdk/core/azure-core/azure/core/pipeline/policies/_retry.py (hunk 1 above)
Comment (johanste) [reply sdk/core/azure-core/azure/core/pipeline/policies/_retry.py]: And the name should be `RetryMode`

Comment (bryevdv) [reply sdk/core/azure-core/azure/core/pipeline/policies/_retry.py]: Agree mode or policy would be better

**Pull request merged**